crates/miden-objects/src/transaction/mod.rs

```diff
@@ -3,19 +3,27 @@ use super::note::{NoteId, Nullifier};
 use super::vm::AdviceInputs;
 use super::{Felt, Hasher, WORD_SIZE, Word, ZERO};
 
+mod advice_inputs;
 mod executed_tx;
 mod inputs;
+mod kernel;
+mod kernel_procedures;
+pub mod memory;
```

Comment:
I would probably move `memory` and maybe some other modules into the `kernel` module.